### PR TITLE
fix(gateway): carve out Authorization from strict header-value cap

### PR DESCRIPTION
**Pull request by cuioss-oliver**
## Summary

Fixes the ~100% failure of the `bearerProxied` benchmark on `main` (run `30552703414`) by carving
`Authorization` out of the strict header-value cap at the non-skippable pre-route floor.

**This branch grew beyond its original brief.** It started as "diagnose and fix the bearer-proxied
benchmark regression". Diagnosis and the phase-6 security audit then surfaced **two further defects of
the same bug class** — a carve-out seeded from the wrong `SecurityConfiguration` baseline — that were
already on `main` and would have silently re-introduced the same rejection one pipeline stage later.
Both are fixed here rather than deferred, because shipping the pre-route carve-out alone would have
produced a fix that a diverging route config undoes.

## Root cause

PLAN-15 (commit `9b8ec4a`) wired the previously **inert** `security_defaults.profile` knob end-to-end.
Before that commit the knob had zero read sites and the gateway-wide inbound baseline was
`SecurityConfiguration.defaults()`; after it, `integration-tests/src/main/docker/sheriff-config/gateway.yaml`'s
declared `profile: strict` actually resolved — so the bearer-proxied route went from effectively
unfiltered to genuinely strict for the first time.

A realm-minted JWT plus its `Bearer ` prefix exceeds strict's **1024-character** `maxHeaderValueLength`
(the builder default is 2048). `BasicChecksStage` — the **non-skippable pre-route floor**, which runs
*before* route selection and *before* bearer validation — therefore rejected ~100% of bearer requests
with `400`. That is exactly the observed benchmark signature: `error_rate 0.999997` at ~10ms latency,
i.e. a fast local rejection rather than an upstream or auth failure.

Why no gate caught it:

- The benchmark lane runs **post-merge only**, so it is not a pre-merge gate.
- `BearerValidationIT` drove only *rejection* scenarios (malformed / missing tokens, all short headers)
  plus one public route — **never a valid bearer through the proxy**. A 21-character
  `Bearer not-a-real-jwt` is nowhere near the cap, so the suite stayed green.

## The fix

An `Authorization` header-value carve-out at the pre-route floor in
`api-sheriff/src/main/java/de/cuioss/sheriff/gateway/pipeline/BasicChecksStage.java`, mirroring the
existing `Cookie`/`Set-Cookie` carve-out at the same header-name dispatch seam. Scoped by **header
name, never by route** — the stage runs before any anchor is resolved, so a per-route lever is
structurally unavailable (and is rejected by ADR-0019).

The budget is **operator-declared, not a hard-coded constant**: a new
`security_defaults.max_authorization_header_value_length` key, default **8192** (the lenient preset
ceiling), added to `SecurityDefaultsConfig.java`, declared in
`api-sheriff/src/main/resources/schema/gateway.schema.json`, and guarded by a boot-time rule in
`ConfigValidator.java` that **refuses a value below the resolved baseline** — so the knob can only
ever widen the cap for `Authorization`, never silently narrow it.

## Two additional defects found and fixed

Both are the same bug class: *a carve-out seeded from the wrong `SecurityConfiguration`.*

**1. `GatewayEdgeRoute.cookieHeaderConfigurationFor` seeded from the DEFAULT preset.** It built its
carve-out from `SecurityConfiguration.builder()` instead of the **resolved** baseline, silently
downgrading `failOnSuspiciousPatterns`, `allowExtendedAscii` and `caseSensitiveComparison` for
`Cookie` values on a strict gateway. ADR-0019's "only the length cap changes" bound is now
**structurally true for the first time**; `doc/adr/0019-Reserved_BFF_paths_bypass_the_url-parameter_value_pipeline_and_the_cookie-header_length_relaxation_is_gateway-wide.adoc`
is updated accordingly.

**2. `ThoroughChecksStage.reRunPipelines` undid BOTH carve-outs one stage later.** It re-validated
every header value under the *route* config with **no carve-out at all** — so for any route whose
`security_filter` diverges from the baseline, the identical `400`-before-bearer-validation regression
reappeared at stage 4. Found by the phase-6 security audit and then **confirmed empirically**, not
merely reasoned about. `ThoroughChecksStage.java` now repeats the same three-way header-name dispatch
with `cap = max(routeCap, budget)`.

## Structural improvement

The 24-component `SecurityConfiguration` copy was factored out of `GatewayEdgeRoute.java` into a
shared `SecurityConfigurations` seam in the ADR-0005-protected `config.model` package
(`api-sheriff/src/main/java/de/cuioss/sheriff/gateway/config/model/SecurityConfigurations.java`). The
component list now exists **exactly once** in production instead of being hand-copied — which is what
made defect (1) above possible in the first place.

## Regression control

- `integration-tests/src/test/java/de/cuioss/sheriff/gateway/integration/BearerValidationIT.java`
  gains a **valid-bearer-through-proxy** scenario: a real token minted from the Keycloak `integration`
  realm via direct grant, driven at the bearer-gated route, asserting the request is **admitted and
  forwarded** (upstream echo present). This is the control that would have caught the regression
  pre-merge. It was **actually run against the live containerised stack**, not merely compiled.
- 11 unit regression tests for the post-route (`ThoroughChecksStage`) carve-out.
- `integration-tests/src/main/docker/certificates/generate-mtls-certificates.sh` no longer rewrites
  three **tracked** files on every run — an unrelated hygiene defect that made every IT run dirty the
  working tree.

## Known gap (stated plainly)

**No integration test covers a bearer-protected route that ALSO declares a `security_filter`.** The
`/secure` anchor in `gateway.yaml` declares none — which is precisely why `BearerValidationIT` could
not have caught the `ThoroughChecksStage` defect, and why that defect needed a security audit to
surface. Unit coverage closes this gap; **IT-level coverage does not.** Adding a diverging-filter
bearer anchor to the IT topology is a real follow-up, deliberately not bundled into this change.

## Changes

- `api-sheriff/.../pipeline/BasicChecksStage.java` — `Authorization` carve-out at the pre-route floor
- `api-sheriff/.../pipeline/ThoroughChecksStage.java` — same carve-out repeated post-route, `max(routeCap, budget)`
- `api-sheriff/.../edge/GatewayEdgeRoute.java` — cookie carve-out re-seeded from the resolved baseline
- `api-sheriff/.../config/model/SecurityConfigurations.java` — new shared 24-component copy seam
- `api-sheriff/.../config/model/SecurityDefaultsConfig.java` — new `max_authorization_header_value_length` key
- `api-sheriff/.../config/validation/ConfigValidator.java` — boot-time below-baseline refusal
- `api-sheriff/src/main/resources/schema/gateway.schema.json` — JSON-schema declaration for the new key
- `integration-tests/.../BearerValidationIT.java` — valid-bearer-through-proxy regression control
- `integration-tests/src/main/docker/sheriff-config/gateway.yaml` — IT gateway document
- `integration-tests/src/main/docker/certificates/generate-mtls-certificates.sh` — stop rewriting tracked files
- `doc/adr/0019-Reserved_BFF_paths_bypass_the_url-parameter_value_pipeline_and_the_cookie-header_length_relaxation_is_gateway-wide.adoc`, `doc/configuration.adoc` — documentation
- Unit tests: `BasicChecksStageTest`, `ThoroughChecksStageTest`, `GatewayEdgeRouteTest`,
  `GatewayEdgeRouteBffWiringTest`, `ConfigValidatorTest`, `ConfigLoaderTest`,
  `ConfigModelContractTest`, `RouteTableBuilderTest`
- `.plan/marshal.json` — steward-maintained artifact riding this PR

## Test Plan

- [x] Quality gate passed (`verify -Ppre-commit`) — zero errors, zero warnings
- [x] Full module test suite green — 1424 tests
- [x] Integration tests green against the live containerised stack — 87 tests
- [x] Pre-merge Performance Benchmark dispatched against this branch as a gate (run `30698473490`)
- [ ] Post-merge Performance Benchmark run — the orchestrator's acceptance check, deliberately outside
      this PR's scope

## Intent

**The problem.** Commit `9b8ec4a` made the `security_defaults.profile` knob live for the first time, so the gateway's inbound baseline became genuinely strict. Strict caps a header value at 1024 characters; a realm JWT plus `Bearer ` exceeds that. `BasicChecksStage` — the non-skippable pre-route floor, which runs before route selection and before bearer validation — therefore rejected essentially every bearer request with 400. No pre-merge gate saw it: the benchmark runs post-merge only, and the bearer IT drove only short rejection tokens.

**The chosen approach.** A carve-out for `Authorization` header values at the same header-name dispatch seam the existing `Cookie` carve-out already uses, scoped by header name and never by route — because at that stage no route is resolved yet, so a per-route lever does not exist. The budget is an operator-declared config key with a boot-time rule refusing any value below the resolved baseline, so the knob can only widen the cap, never narrow it. Relaxing the gateway-wide profile was explicitly rejected as a shipped security regression. The same carve-out is repeated in `ThoroughChecksStage`, because that stage re-validated header values under the route config with no carve-out and thus undid the fix one stage later for any route with a diverging `security_filter`.

**Explicit non-goals.** This change does not relax the gateway-wide

_[Intent truncated — 1392 of 1855 characters shown; full outline in the plan workspace]_

## Summary by CodeRabbit

* **New Features**
  * Added configurable maximum length for `Authorization` header values, defaulting to 8,192 characters.
  * Added gateway-wide handling for `Cookie`, `Set-Cookie`, and `Authorization` header limits while preserving other security checks.
* **Bug Fixes**
  * Invalid security limit configurations are now rejected during startup.
  * mTLS certificates are preserved when still valid, with forced renewal available when needed.
* **Documentation**
  * Documented security limits, header handling, configuration rules, and validation behavior.
* **Tests**
  * Expanded coverage for header validation, route processing, bearer authentication, and certificate handling.

### CODE REVIEW

File: api-sheriff/src/test/java/de/cuioss/sheriff/gateway/config/validation/ConfigValidatorTest.java [hunk 1]
```diff
@@ -461,6 +471,78 @@ void shouldAcceptValidEdgeHardeningBlock() {
...
+                    "is below the resolved baseline max_header_value_length");
+        }
+
+        @Test
+        @DisplayName("Should accept a max_authorization_header_value_length at or above the resolved baseline")
+        void shouldAcceptAuthorizationCapAtOrAboveBaseline() {
+            // Arrange — the matched positive half: 2048 clears the strict baseline, and the same
+            // value is the boundary case 'exactly at the baseline' under a 2048-capped comparison.
+            GatewayConfig above = gatewayWithAuthorizationCap("strict", Optional.of(2048));
+            GatewayConfig atBaseline = gatewayWithAuthorizationCap("strict", Optional.of(1024));
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply api-sheriff/src/test/java/de/cuioss/sheriff/gateway/config/validation/ConfigValidatorTest.java]: Fixed - allocated TASK-008. Confirmed against the worktree: shouldAcceptAuthorizationCapAtOrAboveBaseline resolves the strict baseline (1024), so the boundary case is the atBaseline fixture on line 514, not the above fixture on line 513, and there is no 2048-capped comparison anywhere in this test - that phrasing is leftover from the preceding lenient-baseline test at lines 495-497. You are also right that the assertions and their failure messages at lines 523-525 are already correct, so the change is comment-only: no fixture and no assertion is touched.

File: doc/adr/0019-Reserved_BFF_paths_bypass_the_url-parameter_value_pipeline_and_the_cookie-header_length_relaxation_is_gateway-wide.adoc [hunk 2]
```diff
@@ -134,14 +140,165 @@ Consequences of the amendment:
...
+
+* *By header.* Only `Authorization` values are affected. Every other header keeps the resolved
+  baseline cap.
+* *By validator.* Only the length cap changes — see the correction below, which is what makes this
+  claim true.
+* *By deployment.* *Unconditional.* Unlike the cookie carve-out, which exists only in an active
+  cookie-mode BFF, this one is present on every gateway, because a bearer-capable gateway is every
+  gateway. There is no mode axis to contain it. That is a genuinely larger residual exposure than
+  decision (b)'s original third axis, and it is accepted knowingly: the alternative is a gateway that
+  cannot serve bearer traffic at all.
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply doc/adr/0019-Reserved_BFF_paths_bypass_the_url-parameter_value_pipeline_and_the_cookie-header_length_relaxation_is_gateway-wide.adoc:185]: Fixed - allocated TASK-009. Verified against the record: the Risks bullet at lines 331-334 still enumerates 'one header pair, one validator, one deployment mode', while the amendment at lines 181-185 states the Authorization carve-out is Unconditional and is 'a genuinely larger residual exposure than decision (b)'s original third axis'. Two of the three axes are therefore stale - it is two header names now, and one of them has no deployment axis at all - and only 'one validator' survives unchanged. The existing cross-reference at lines 171-172 addresses solely the different-validator half and leaves the other two. Your point about the consequence is the one that decided this: the record's own stated value at lines 313-314 is that residual exposure is 'explicitly enumerable rather than emergent, which is what makes it reviewable', and a reader who consults only the Risks section gets an understated blast radius for a security relaxation. We are taking your suggested remedy rather than editing the bullet: the correction goes into the existing amendment as added prose, following this record's own precedent at lines 193-197 where the amendment retires the key-placement risk for one carve-out in prose. ADRs here are amended, never rewritten.

File: doc/configuration.adoc [hunk 3]
```diff
@@ -988,6 +988,99 @@ The explicit *not-a-WAF* boundary: allowlists and limits, no attack-signature ma
...
+
+| cookie value length / cookie count
+| 1024 / 10
+| 8192 / 50
+|===
+
+All length limits are measured in *characters* -- the `cui-http` unit -- not bytes. The presets also
+differ in their non-length validators: `strict` normalizes Unicode, compares case-sensitively, fails
+on suspicious patterns, and permits neither double encoding, control characters, nor extended ASCII;
+`lenient` inverts each of those. Neither preset ever permits null bytes.
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply doc/configuration.adoc:1038]: Not actioned - recorded as residue rather than dismissed. The drift risk you name is real and this project has already been bitten by it: commit 9b8ec4a moved the resolved baseline from the cui-http builder default to strict(), left three documentation sites stale, and produced the ~100% bearer rejection this PR exists to fix. Two things weigh against the proposed pinning test here. First, it asserts a third-party library's own constants rather than this project's behaviour, so it goes red on every intended cui-http upgrade and the corrective action it forces is the doc edit that upgrade needed anyway - a genuine forcing function, but not a free one, and it puts a dependency's release cadence on our build's critical path. Second and decisively, a new test class asserting library presets is scope outside all four of this plan's deliverables (diagnose, fix+docs+ADR, regression IT, lane proof), on a branch already at loop-back iteration 2 for a security fix with partial bot review coverage. What IS being fixed this cycle is the concrete instance behind your concern: TASK-009 corrects doc/configuration.adoc line 1409, which states a 2048-character default that no code path produces, and re-anchors the neighbouring bullets on the preset-limits table instead of on inline absolutes. The generic pinning mechanism is recorded for follow-up rather than added to a security-fix branch at the end of its cycle.

File: doc/configuration.adoc [hunk 4]
```diff
@@ -988,6 +988,99 @@ The explicit *not-a-WAF* boundary: allowlists and limits, no attack-signature ma
...
+
+[[_pre_route_header_carve_outs]]
+==== Pre-route header-name carve-outs
+
+The *non-skippable pre-route floor* measures every inbound header value against the resolved
+baseline `max_header_value_length` above. Two header names are carved out of that cap, each by a
+dedicated value pipeline selected on the header *name*. Both carve-outs are declared on gateway-wide
+blocks, because the floor runs *before route selection* -- no anchor is resolved at that point, so
+neither relaxation can be scoped per route or per anchor
+(link:adr/0019-Reserved_BFF_paths_bypass_the_url-parameter_value_pipeline_and_the_cookie-header_length_relaxation_is_gateway-wide.adoc[ADR-0019]).
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply doc/configuration.adoc:1048]: Fixed - allocated TASK-009. Confirmed against the worktree, and it is worse than one stale figure: the document contradicts itself. Line 1043 and the new carve-out table at lines 1054-1058 both use the resolved-baseline framing, the preset table at lines 1018-1020 gives 1024 strict and 8192 lenient, and line 1409 still states a fixed 2048 - the dropped cui-http builder default, which neither shipped preset produces. Line 1411 ('Every other header keeps the default') carries the same stale framing in the very next bullet, so TASK-009 corrects both and cross-references the preset-limits section rather than restating numbers inline, which is what stops this class of drift recurring. This is the third instance of exactly this defect on the branch: decision.log 2a44f1 already caught and fixed two other stale 2048 sites (the BasicChecksStage javadoc and the cookieHeaderConfigurationFor javadoc) and this site was missed by that sweep.

File: integration-tests/src/main/docker/certificates/generate-mtls-certificates.sh [hunk 5]
```diff
@@ -22,7 +29,58 @@ CERT_DIR="${SCRIPT_DIR}"
...
+        return 0
+    fi
+
+    local artifact
+    for artifact in "${RETAINED_ARTIFACTS[@]}"; do
+        if [[ ! -f "${artifact}" ]]; then
+            REGENERATION_REASON="missing $(basename "${artifact}")"
+            return 0
+        fi
+    done
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply integration-tests/src/main/docker/certificates/generate-mtls-certificates.sh:64]: Partially fixed - allocated TASK-010. You are right about the gap: regeneration_needed() checks presence for all three artifacts and openssl x509 -checkend for the CA only, so the two PKCS#12 keystores are never verified loadable, and line 78 nonetheless announces 'already present and valid'. TASK-010 corrects that message to state what was actually checked, so a future MtlsHandshakeIT keystore failure is not debugged from a message that falsely rules out the material - which is the part of your finding that is genuinely load-bearing. We are deliberately NOT adding openssl pkcs12 load validation. TASK-006 scoped that out when this guard was written, and although its recorded reason (avoiding the p12 passwords) does not actually hold - they are non-secret literals in the same file at lines 8-11 - the proportionality argument does. All three artifacts are tracked in git, so a zero-byte or truncated keystore is not a state a checkout produces; when it does occur the failure is loud and local (MtlsHandshakeIT errors on keystore load) and --force is the documented one-step recovery. Two extra openssl invocations on every pre-integration-test run is not a proportionate price for that, whereas making the message truthful costs nothing at runtime. Recorded so the trade is visible rather than silent.

File: integration-tests/src/main/docker/certificates/generate-mtls-certificates.sh [hunk 6]
```diff
@@ -22,7 +29,58 @@ CERT_DIR="${SCRIPT_DIR}"
...
+
+    # All three artifacts are issued in one pass with the same validity, so the CA certificate's
+    # expiry is representative of the set — and unlike the PKCS#12 keystores it can be inspected
+    # without a password. A non-zero exit here also covers unreadable/corrupt material.
+    if ! openssl x509 -checkend "${RENEWAL_MARGIN_SECONDS}" -noout -in "${CERT_DIR}/mtls-client-ca.crt" >/dev/null; then
+        REGENERATION_REASON="mtls-client-ca.crt expires within $((RENEWAL_MARGIN_SECONDS / 86400)) days"
+        return 0
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply integration-tests/src/main/docker/certificates/generate-mtls-certificates.sh:75]: Fixed - allocated TASK-010, as a reason-string correction rather than a control-flow split. You are right that a non-zero openssl x509 -checkend also covers unreadable and unparseable material, so the reason 'mtls-client-ca.crt expires within N days' can be wrong about the cause and is echoed as such at line 83. Two qualifications on the shape of the fix. The branch is deliberate rather than an oversight - the comment at line 68 already records that it covers unreadable/corrupt material too - and regeneration is the correct action for every cause the branch catches, so the behaviour is right and stays byte-for-byte. What was actually wrong is only the diagnostic string, which sends a reader down the expiry path for corrupt material. TASK-010 therefore widens the reason to name unreadable, unparseable and expiring, keeping the single branch and the single openssl invocation rather than parsing the certificate first.

File: api-sheriff/src/test/java/de/cuioss/sheriff/gateway/config/validation/ConfigValidatorTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply api-sheriff/src/test/java/de/cuioss/sheriff/gateway/config/validation/ConfigValidatorTest.java]: Fixed - allocated TASK-008. Confirmed against the worktree: shouldAcceptAuthorizationCapAtOrAboveBaseline resolves the strict baseline (1024), so the boundary case is the atBaseline fixture on line 514, not the above fixture on line 513, and there is no 2048-capped comparison anywhere in this test - that phrasing is leftover from the preceding lenient-baseline test at lines 495-497. You are also right that the assertions and their failure messages at lines 523-525 are already correct, so the change is comment-only: no fixture and no assertion is touched.

File: doc/adr/0019-Reserved_BFF_paths_bypass_the_url-parameter_value_pipeline_and_the_cookie-header_length_relaxation_is_gateway-wide.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply doc/adr/0019-Reserved_BFF_paths_bypass_the_url-parameter_value_pipeline_and_the_cookie-header_length_relaxation_is_gateway-wide.adoc:185]: Fixed - allocated TASK-009. Verified against the record: the Risks bullet at lines 331-334 still enumerates 'one header pair, one validator, one deployment mode', while the amendment at lines 181-185 states the Authorization carve-out is Unconditional and is 'a genuinely larger residual exposure than decision (b)'s original third axis'. Two of the three axes are therefore stale - it is two header names now, and one of them has no deployment axis at all - and only 'one validator' survives unchanged. The existing cross-reference at lines 171-172 addresses solely the different-validator half and leaves the other two. Your point about the consequence is the one that decided this: the record's own stated value at lines 313-314 is that residual exposure is 'explicitly enumerable rather than emergent, which is what makes it reviewable', and a reader who consults only the Risks section gets an understated blast radius for a security relaxation. We are taking your suggested remedy rather than editing the bullet: the correction goes into the existing amendment as added prose, following this record's own precedent at lines 193-197 where the amendment retires the key-placement risk for one carve-out in prose. ADRs here are amended, never rewritten.

File: doc/configuration.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply doc/configuration.adoc:1038]: Not actioned - recorded as residue rather than dismissed. The drift risk you name is real and this project has already been bitten by it: commit 9b8ec4a moved the resolved baseline from the cui-http builder default to strict(), left three documentation sites stale, and produced the ~100% bearer rejection this PR exists to fix. Two things weigh against the proposed pinning test here. First, it asserts a third-party library's own constants rather than this project's behaviour, so it goes red on every intended cui-http upgrade and the corrective action it forces is the doc edit that upgrade needed anyway - a genuine forcing function, but not a free one, and it puts a dependency's release cadence on our build's critical path. Second and decisively, a new test class asserting library presets is scope outside all four of this plan's deliverables (diagnose, fix+docs+ADR, regression IT, lane proof), on a branch already at loop-back iteration 2 for a security fix with partial bot review coverage. What IS being fixed this cycle is the concrete instance behind your concern: TASK-009 corrects doc/configuration.adoc line 1409, which states a 2048-character default that no code path produces, and re-anchors the neighbouring bullets on the preset-limits table instead of on inline absolutes. The generic pinning mechanism is recorded for follow-up rather than added to a security-fix branch at the end of its cycle.

File: doc/configuration.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply doc/configuration.adoc:1048]: Fixed - allocated TASK-009. Confirmed against the worktree, and it is worse than one stale figure: the document contradicts itself. Line 1043 and the new carve-out table at lines 1054-1058 both use the resolved-baseline framing, the preset table at lines 1018-1020 gives 1024 strict and 8192 lenient, and line 1409 still states a fixed 2048 - the dropped cui-http builder default, which neither shipped preset produces. Line 1411 ('Every other header keeps the default') carries the same stale framing in the very next bullet, so TASK-009 corrects both and cross-references the preset-limits section rather than restating numbers inline, which is what stops this class of drift recurring. This is the third instance of exactly this defect on the branch: decision.log 2a44f1 already caught and fixed two other stale 2048 sites (the BasicChecksStage javadoc and the cookieHeaderConfigurationFor javadoc) and this site was missed by that sweep.

File: integration-tests/src/main/docker/certificates/generate-mtls-certificates.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply integration-tests/src/main/docker/certificates/generate-mtls-certificates.sh:64]: Partially fixed - allocated TASK-010. You are right about the gap: regeneration_needed() checks presence for all three artifacts and openssl x509 -checkend for the CA only, so the two PKCS#12 keystores are never verified loadable, and line 78 nonetheless announces 'already present and valid'. TASK-010 corrects that message to state what was actually checked, so a future MtlsHandshakeIT keystore failure is not debugged from a message that falsely rules out the material - which is the part of your finding that is genuinely load-bearing. We are deliberately NOT adding openssl pkcs12 load validation. TASK-006 scoped that out when this guard was written, and although its recorded reason (avoiding the p12 passwords) does not actually hold - they are non-secret literals in the same file at lines 8-11 - the proportionality argument does. All three artifacts are tracked in git, so a zero-byte or truncated keystore is not a state a checkout produces; when it does occur the failure is loud and local (MtlsHandshakeIT errors on keystore load) and --force is the documented one-step recovery. Two extra openssl invocations on every pre-integration-test run is not a proportionate price for that, whereas making the message truthful costs nothing at runtime. Recorded so the trade is visible rather than silent.

File: integration-tests/src/main/docker/certificates/generate-mtls-certificates.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply integration-tests/src/main/docker/certificates/generate-mtls-certificates.sh:75]: Fixed - allocated TASK-010, as a reason-string correction rather than a control-flow split. You are right that a non-zero openssl x509 -checkend also covers unreadable and unparseable material, so the reason 'mtls-client-ca.crt expires within N days' can be wrong about the cause and is echoed as such at line 83. Two qualifications on the shape of the fix. The branch is deliberate rather than an oversight - the comment at line 68 already records that it covers unreadable/corrupt material too - and regeneration is the correct action for every cause the branch catches, so the behaviour is right and stays byte-for-byte. What was actually wrong is only the diagnostic string, which sends a reader down the expiry path for corrupt material. TASK-010 therefore widens the reason to name unreadable, unparseable and expiring, keeping the single branch and the single openssl invocation rather than parsing the certificate first.

File: api-sheriff/src/main/java/de/cuioss/sheriff/gateway/config/validation/ConfigValidator.java [hunk 7]
```diff
@@ -224,6 +228,51 @@ private static void validateEdgeHardening(GatewayConfig gateway, List<ConfigErro
...
+                    if (declared < baseline) {
+                        errors.add(new ConfigError(GATEWAY_FILE, SECURITY_DEFAULTS_AUTHORIZATION_POINTER,
+                                ("max_authorization_header_value_length %d is below the resolved baseline "
+                                        + "max_header_value_length %d; the key raises the pre-route cap for "
+                                        + "Authorization values only and can never lower it — tighten through "
+                                        + "security_defaults.profile or a route security_filter instead")
+                                        .formatted(declared, baseline)));
+                    }
+                });
+    }
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply api-sheriff/src/main/java/de/cuioss/sheriff/gateway/config/validation/ConfigValidator.java:285]: Confirmed against the source, and it is worse than the comment states — neither named remedy can tighten an Authorization value. A route security_filter cannot, because carveOutConfigurationFor floors the cap at max(routeConfig.maxHeaderValueLength(), budget); security_defaults.profile cannot either, because the budget resolves from this key or the fixed 8192 default and does not track the profile. Will be addressed by TASK-012, message text only. We are deliberately NOT taking the alternative of exposing a downward path for the Authorization budget: the loud boot refusal is an intentional standing decision on this plan, so the message gets corrected rather than the enforcement relaxed.

File: integration-tests/src/test/java/de/cuioss/sheriff/gateway/integration/BearerValidationIT.java [hunk 8]
```diff
@@ -81,4 +106,56 @@ void publicRouteDoesNotRequireBearer() {
...
+                .when()
+                .get("/secure/get")
+                .then()
+                .statusCode(200)
+                .extract();
+
+        // Assert — a non-null echoed method is the observable proof the request was not merely
+        // accepted but actually forwarded to the go-httpbin upstream.
+        assertEquals("GET", response.path("method"),
+                "an admitted bearer request must reach the upstream and echo its method");
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply integration-tests/src/test/java/de/cuioss/sheriff/gateway/integration/BearerValidationIT.java:136]: Agreed, and weighted above its Minor label — this IT is the regression control the whole change exists to add, so an unasserted header length lets it pass green while never exercising the carve-out at all. Will be addressed by TASK-011: the Authorization value is built once, asserted length() > 1024 against the strict baseline, then sent. See the follow-up commit on this branch.
